src/main/java/org/openrewrite/java/testing/junit6/MinimumJreConditions.java

```diff
@@ -0,0 +1,412 @@
+                        maybeRemoveImport(JRE_IMPORT);
+                        return null;
+                    }
+                    if (compareVersions(range.getMin(), javaVersion) <= 0 && range.getMaxNotation() == null) {
+                        // The test is enabled for all versions from `javaVersion till end -> remove annotation
+                        m = removeAnnotation(m, ENABLED_FOR_JRE_RANGE, ctx);
+                    } else {
+                        m = updateRangeStart(m, ENABLED_FOR_JRE_RANGE, range, ctx);
+                    }
+                }
+                if (disabledOnJreRange.isPresent()) {
+                    Range range = disabledOnJreRange.get();
+                    if (compareVersions(range.getMax(), javaVersion) < 0) {
+                        m = removeAnnotation(m, DISABLED_FOR_JRE_RANGE, ctx);
+                    } else if (compareVersions(range.getMin(), javaVersion) <= 0 && range.getMaxNotation() == null) {
+                        // The test is disabled for all versions from `javaVersion` till end -> remove test
+                        maybeRemoveImport(JRE_IMPORT);
+                        return null;
+                    } else {
+                        m = updateRangeStart(m, DISABLED_FOR_JRE_RANGE, range, ctx);
+                    }
+                }
+
+                replaceSingleVersionRangeWithEquivalentAnnotation(enabledOnJreRange, ENABLED_FOR_JRE_RANGE, ENABLED_ON_JRE);
+                replaceSingleVersionRangeWithEquivalentAnnotation(disabledOnJreRange, DISABLED_FOR_JRE_RANGE, DISABLED_ON_JRE);
+
+                if (m != method) {
+                    return simplifySingleValueAnnotationAttributeArrays(m, prefix);
+                }
+                return method;
+            }
+
+            private J.MethodDeclaration removeAnnotation(J.MethodDeclaration m, String annotationType, ExecutionContext ctx) {
+                RemoveAnnotation removeAnnotation = new RemoveAnnotation("@" + annotationType);
+                maybeRemoveImport(JRE_IMPORT);
+                return removeAnnotation.getVisitor().visitMethodDeclaration(m, ctx);
+            }
+
+            private J.MethodDeclaration updateAnnotationVersions(J.MethodDeclaration m, Optional<List<String>> versions, String annotationType, ExecutionContext ctx) {
+                if (versions.isPresent() && versions.get().stream().anyMatch(v -> compareVersions(v, javaVersion) < 0)) {
+                    // Update the annotation to only include JRE versions that are greater than or equal to the specified version
+                    AddOrUpdateAnnotationAttribute updatedVersions = new AddOrUpdateAnnotationAttribute(annotationType, "versions", versions.get().stream().filter(v -> compareVersions(v, javaVersion) >= 0).collect(joining(", ")), null, false, false);
+                    m = (J.MethodDeclaration) updatedVersions.getVisitor().visit(m, ctx, getCursor().getParent());
```

Comment:
```suggestion
                    m = (J.MethodDeclaration) updatedVersions.getVisitor().visitNonNull(m, ctx, getCursor().getParent());
```

Check also all the other `.visit`  methods!